specification/cognitiveservices/data-plane/ContentSafety/stable/2023-10-01/contentsafety.json

```diff
@@ -964,8 +986,8 @@
         },
         "text": {
           "type": "string",
-          "description": "BlocklistItem content.",
-          "maxLength": 128
+          "description": "BlocklistItem content. The length is counted using Unicode code point.",
+          "maxLength": 1000
```

Comment:
Should not change maxLength in released API definitions.

Reply:
How can we just apply the new maxLength for the new version and keep the original in released version?=

Reply:
Also I don't quite understand why a larger maxLength is considered as a breaking change. With a larger maxLength, the existing client implementation won't be affected.

Reply:
I think Mike's concern is that this change applies to 2023-10-01 api-version, not the new 2024-09-01. And the general rule is there should be no API change to existing api-version.

What is the maxLength for this API in 2023-10-01?

Currently I am not aware of a way to versioning the decorator in TypeSpec. Maybe you can check in TypeSpec Discussion channel.
An ugly fallback (if no other alternative) would be `@remove` then `@add`.

Reply:
maxLength in 2023-10-01 is 128, and we're going to increase it to 1000 in 2024-09-01.
@mikekistler , can you help advise further?

Reply:
Please come to Breaking Change office hours today at 10AM PT so we can discuss.

[Breaking Change Office Hours](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/apex_support)

Reply:
@weidongxu-microsoft , thank you for the suggestion. I tried below but  both not work. Can you help advise further how to use the remove/add to do the fallback? I posted [this question](https://teams.microsoft.com/l/message/19:906c1efbbec54dc8949ac736633e6bdf@thread.skype/1723041785347?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1723041785347&teamName=Azure%20SDK&channelName=TypeSpec%20Discussion&createdTime=1723041785347) into the TypeSpec Discussion as well. 

1.
  @removed(ContentSafety.Versions.v2024_09_01)
  @doc("BlocklistItem content. The length is counted using Unicode code point.")
  @maxLength(128)
  text: string;

  @added(ContentSafety.Versions.v2024_09_01)
  @doc("BlocklistItem content. The length is counted using Unicode code point.")
  @maxLength(1000)
  text: string;

2.
  @removed(ContentSafety.Versions.v2024_09_01)
  @added(ContentSafety.Versions.v2024_09_01)
  @doc("BlocklistItem content. The length is counted using Unicode code point.")
  @maxLength(1000)
  text: string;


> I think Mike's concern is that this change applies to 2023-10-01 api-version, not the new 2024-09-01. And the general rule is there should be no API change to existing api-version.
> 
> What is the maxLength for this API in 2023-10-01?
> 
> Currently I am not aware of a way to versioning the decorator in TypeSpec. Maybe you can check in TypeSpec Discussion channel. An ugly fallback (if no other alternative) would be `@remove` then `@add`.

Reply:
I assume #1 would give you a "duplicate entity" error?

The answer from Timothee and Mark is better than mine (they are the expert on TypeSpec).

I guess you would have to discuss this with Mike on breaking change office hour. Do show this discussion in TypeSpec Discussion.

Reply:
Reverted it to 128.